ext/posix/posix.stub.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+function posix_kill(int $pid, int $sig): bool {}
+
+function posix_getpid(): int {}
+
+function posix_getppid(): int {}
+
+function posix_getuid(): int {}
+
+function posix_setuid(int $uid): bool {}
+
+function posix_geteuid(): int {}
+
+#ifdef HAVE_SETEUID
+function posix_seteuid(int $uid): bool {}
+#endif
+
+function posix_getgid(): int {}
+
+function posix_setgid(int $gid): bool {}
+
+function posix_getegid(): int {}
+
+#ifdef HAVE_SETEGID
+function posix_setegid(int $gid): bool {}
+#endif
+
+#ifdef HAVE_GETGROUPS
+/** @return array|false */
+function posix_getgroups() {}
+#endif
+
+#ifdef HAVE_GETLOGIN
+/** @return string|false */
+function posix_getlogin() {}
+#endif
+
+function posix_getpgrp(): int {}
+
+#ifdef HAVE_SETSID
+function posix_setsid(): int {}
+#endif
+
+function posix_setpgid(int $pid, int $pgid): bool {}
+
+#ifdef HAVE_GETPGID
+/** @return int|false */
+function posix_getpgid(int $pid) {}
+#endif
+
+#ifdef HAVE_GETSID
+/** @return int|false */
+function posix_getsid(int $pid) {}
+#endif
+
+/** @return array|false */
+function posix_uname() {}
+
+/** @return array|false */
+function posix_times() {}
+
+
+#ifdef HAVE_CTERMID
+/** @return string|false */
+function posix_ctermid() {}
+#endif
+
+/**
+ * @param int|resource $fd
```

Comment:
This is not quite right; $fd can be an arbitrary value which is either a resource, or is *converted* to int (the latter regardless of strict_mode). Basically the same for `posix_ttyname` below. I'm taking the liberty to fix these minor issues myself.